tutorials/bios-boot-tutorial/bios-boot-tutorial.py

```diff
@@ -293,9 +296,53 @@ def stepCreateTokens():
     run(args.cleos + 'push action eosio.token issue \'["eosio", "%s", "memo"]\' -p eosio' % intToCurrency(totalAllocation))
     sleep(1)
 def stepSetSystemContract():
+    # Nearly all of the protocol upgrade features introduced in v1.8 first require a special protocol 
+    # feature (codename PREACTIVATE_FEATURE) to be activated and for an updated version of the system 
+    # contract that utilizes the functionality introduced by that feature to be deployed. 
+
+    # activate PREACTIVATE_FEATURE before installing eosio.bios
+    retry('curl -X POST http://127.0.0.1:%d' % args.http_port + 
+        '/v1/producer/schedule_protocol_feature_activations ' +
+        '-d \'{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}\'')
+    sleep(5)
+
+    # install eosio.system
     retry(args.cleos + 'set contract eosio ' + args.contracts_dir + '/eosio.system/')
     sleep(1)
+
+    # activate remaining features, uncomment any of the lines below
+    # # GET_SENDER
+    # retry(args.cleos + 'push action eosio activate \'["f0af56d2c5a48d60a4a5b5c903edfb7db3a736a94ed589d0b797df33ff9d3e1d"]\' -p eosio')
+    # sleep(1)
+    # # FORWARD_SETCODE
+    # retry(args.cleos + 'push action eosio activate \'["2652f5f96006294109b3dd0bbde63693f55324af452b799ee137a81a905eed25"]\' -p eosio')
+    # sleep(1)
+    # # ONLY_BILL_FIRST_AUTHORIZER
+    # retry(args.cleos + 'push action eosio activate \'["8ba52fe7a3956c5cd3a656a3174b931d3bb2abb45578befc59f283ecd816a405"]\' -p eosio')
+    # sleep(1)
+    # # RESTRICT_ACTION_TO_SELF
+    # retry(args.cleos + 'push action eosio activate \'["ad9e3d8f650687709fd68f4b90b41f7d825a365b02c23a636cef88ac2ac00c43"]\' -p eosio')
+    # sleep(1)
+    # # DISALLOW_EMPTY_PRODUCER_SCHEDULE
+    # retry(args.cleos + 'push action eosio activate \'["68dcaa34c0517d19666e6b33add67351d8c5f69e999ca1e37931bc410a297428"]\' -p eosio')
+    # sleep(1)
+    #  # FIX_LINKAUTH_RESTRICTION
+    # retry(args.cleos + 'push action eosio activate \'["e0fb64b1085cc5538970158d05a009c24e276fb94e1a0bf6a528b48fbc4ff526"]\' -p eosio')
+    # sleep(1)
+    #  # REPLACE_DEFERRED
+    # retry(args.cleos + 'push action eosio activate \'["ef43112c6543b88db2283a2e077278c315ae2c84719a8b25f25cc88565fbea99"]\' -p eosio')
+    # sleep(1)
+    # # NO_DUPLICATE_DEFERRED_ID
+    # retry(args.cleos + 'push action eosio activate \'["4a90c00d55454dc5b059055ca213579c6ea856967712a56017487886a4d4cc0f"]\' -p eosio')
+    # sleep(1)
+    # # ONLY_LINK_TO_EXISTING_PERMISSION
+    # retry(args.cleos + 'push action eosio activate \'["1a99a59d87e06e09ec5b028a9cbb7749b4a5ad8819004365d02dc4379a8b7241"]\' -p eosio')
+    # sleep(1)
+    # # RAM_RESTRICTIONS
+    # retry(args.cleos + 'push action eosio activate \'["4e7bf348da00a945489b2a681749eb56f5de00b900014e137ddae39f48f69d67"]\' -p eosio')
+    # sleep(1)
```

Comment:
Seems by default we should enable all of these.

Reply:
I do not understand why 'by default' it should enable all of them, can you please explain more?
The way I see it, we leave it like that and the user is choosing which one to uncomment based on which one she wants to experiment with.

Reply:
I guess I see the bios boot script as setting up minimum current chain. We expect/desire all these features to be enabled on eosio chains. They are simply features that required hard-fork to enable. If we started a new chain we would start it with all these enabled.

Reply:
ah, ok, I did not get it that way, that we expect/desire all these features to be enabled on eosio chains. alrighty then, I will uncomment and have them enabled by default. thank you @heifner 👍